posts/tools/test-frame-introduction.md

```diff
@@ -0,0 +1,174 @@
+
+### How I can use it?
+
+During the development, the Test-Frame library is being pushed to GitHub's Maven repository.
+So in case you would like to try the Test-Frame in your test suite before it is released, you can do that by including
+the GitHub's Maven repository and the Test-Frame dependency in your `pom.xml`:
+
+```xml
+    <repositories>
+        <repository>
+            <id>test-frame</id>
+            <name>GitHub Apache Maven Packages</name>
+            <url>https://maven.pkg.github.com/skodjob/test-frame</url>
+        </repository>
+    </repositories>
+    <dependencies>
+        <dependency>
+            <groupId>io.skodjob</groupId>
+            <artifactId>test-frame-common</artifactId>
+            <version>0.1.0-SNAPSHOT</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+```
+where the `test-frame-common` module contains the core functionality - like the `KubeResourceManager`.
+Additionally, there are more modules that you can use and explore in our [GitHub repository](https://github.com/skodjob/test-frame).
+
+Because we are pulling a dependency from the GitHub Maven repository, we need to provide an additional configuration in the `settings.xml`:
+
+```xml
+<settings>
+    <servers>
+        <server>
+            <id>test-frame</id>
+            <username>x-access-token</username>
+            <password>${env.GITHUB_TOKEN}</password>
+        </server>
+    </servers>
+</settings>
+```
+
+Where the `id` of the server has to match with the `id` of the repository in the `pom.xml`.
+
+Once you are all set, you can try to experiment with the library in the tests by annotating the test classes with
+`@ResourceManager` annotation, determining that the `KubeResourceManager` should manage the resources created in the tests.
+
+```java
+@ResourceManager
+class NamespaceCreationST {
+    
+    @Test
+    void testThatNamespaceExists() {
+        KubeResourceManager.getInstance().createResourceWithWait(
+                new NamespaceBuilder().withNewMetadata().withName("test").endMetadata().build());
+        assertNotNull(KubeResourceManager.getKubeCmdClient().get("namespace", "test"));
+    }
+}
+```
+
+In this example, the Namespace with the name `test` is created in the test case `testThatNamespaceExists`, and after the test
+is finished, it is also deleted by the `KubeResourceManager`.
+This will ensure that the Namespace will not affect any other tests, which will start again in a clean environment.
+
+## Conclusion
+
+The process of writing tests with all kinds of helper classes and methods can be a repetitive task filled with duplicates and
+redundancy when it comes to a project containing multiple repositories that need testing.
+But Test-Frame, a library providing common helper methods for work with tests and objects running on Kubernetes, is here
+to help you with it.
+
+Let us know how you like the library or what would you like to see next.
+Your feedback is welcome!
```

Comment:
I think we could also add a note that we welcome all the contributions or something like that.